core/domain/suggestion_jobs_one_off.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import ast
+import logging
+
+import bs4
+from core import jobs
+from core.domain import exp_domain
+from core.domain import html_validation_service
+from core.domain import question_domain
+from core.domain import state_domain
+from core.domain import suggestion_services
+from core.platform import models
+
+(suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
+
+
+class SuggestionMathRteAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that checks for existence of math RTEs in the suggestions."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        html_string_list = []
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        if (suggestion.suggestion_type ==
+                suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+            suggestion_dict = suggestion.to_dict()
+            html_string_list.append(
+                suggestion_dict['change']['translation_html'])
+            html_string_list.append(
+                suggestion_dict['change']['content_html'])
+        elif (suggestion.suggestion_type ==
+              suggestion_models.SUGGESTION_TYPE_ADD_QUESTION):
+            suggestion_dict = suggestion.to_dict()
+            interaction_object = (
+                state_domain.InteractionInstance.from_dict(
+                    suggestion_dict['change']['question_dict'][
+                        'question_state_data']['interaction']))
+            html_string_list = (
+                html_string_list +
+                interaction_object.get_all_html_content_strings())
+            written_translations_dict = (
+                suggestion_dict['change']['question_dict'][
+                    'question_state_data']['written_translations'])
+            for (content_id, language_code_to_written_translation) in (
+                    written_translations_dict['translations_mapping'].items()):
+                for language_code in (
+                        language_code_to_written_translation.keys()):
+                    written_translation_html = (
+                        written_translations_dict['translations_mapping'][
+                            content_id][language_code]['html'])
+                    html_string_list.append(written_translation_html)
+        html_string = ''.join(html_string_list)
+        soup = bs4.BeautifulSoup(
+            html_string.encode(encoding='utf-8'), 'html.parser')
+        if soup.findAll(name='oppia-noninteractive-math'):
+            yield ('Suggestion with Math', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield ('%d suggestions have Math RTEs in them with IDs.' % len(values),
+               values)
+
+
+class SuggestionMathMigrationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
```

Comment:
Can we move this to the line above? (Following max 80 chars in a line)

Reply:
done